src/java/org/apache/cassandra/config/DurationSpec.java

```diff
@@ -62,77 +64,72 @@ public Duration(String value)
         }
         else if(matcherDouble.find())
         {
-            quantity =(long) Double.parseDouble(matcherDouble.group(1));
+            quantity = Math.round(Double.parseDouble(matcherDouble.group(1)));
```

Comment:
There are some things like `0.3h` that we'll round down to zero, right? I think `commitlog_sync_group_window_in_ms` is the only thing that uses the double converter anyway, so if we do keep it, perhaps we only allow `ms`?

Reply:
I just noticed with the currently committed `Duration` class the guys gave the chance to add ttl in double, IMHO this is a bug. I will fix it as part of this ticket.
I will change it and leave only ' commitlog_sync_group_window_in_ms` to be possible to be `double` in `ms` and rounded when stored as `long quantity` during conversion to the new spec. The new `commitlog_sync_group_window` should be allowed to be only `long`, I remember discussing this before whether to keep the double.